### PR TITLE
[1.21] vendor: bump runc to 1.0.0

### PR DESCRIPTION
**Pull request by kolyshkin**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Bumping runc dependency to 1.0.0, fixing a few issues.

This is a backport of https://github.com/kubernetes/kubernetes/pull/102508 to release-1.21 branch.

#### Which issue(s) this PR fixes:
Fixes: #102676

#### Special notes for your reviewer:

I am not backporting the second commit from PR #102508 as it's not important (a cleanup).

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

n/a

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -48,9 +48,9 @@ require (
 	github.com/gogo/protobuf v1.3.2
 	github.com/golang/groupcache v0.0.0-20200121045136-8c9f03a8e57e
 	github.com/golang/mock v1.4.4
-	github.com/golang/protobuf v1.4.3
+	github.com/golang/protobuf v1.5.0
```
Comment (liggitt) [inline go.mod:51]: uh oh... was this really required on the runc v1.0.0 bump? I need to check if this is in the set of problematic grpc/protobuf version crossings that https://github.com/kubernetes/kubernetes/pull/100488 was required to fix

Comment (liggitt) [reply go.mod:51]: yeah... runc 1.0.0 bumped to google.golang.org/protobuf@v1.26.0 which requires this

**Comment by liggitt** [comment]
the new version of proto libraries brought deprecations with them (see https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103392/pull-kubernetes-verify/1410422073385291776)

see fix in master in https://github.com/kubernetes/kubernetes/pull/100488/commits/ff2c61424598924a681e66d71e8deeb21b05101e

**Comment by liggitt** [comment]
lgtm once CI is clean

**Comment by liggitt** [comment]
is there a CI job demonstrating https://github.com/kubernetes/kubernetes/issues/102676 we can see turn green with this?

File: go.mod (hunk 1 above)
Comment (kolyshkin) [reply go.mod:51]: > the new version of proto libraries brought deprecations with them (see https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/103392/pull-kubernetes-verify/1410422073385291776)
> 
> see fix in master in [ff2c614](https://github.com/kubernetes/kubernetes/commit/ff2c61424598924a681e66d71e8deeb21b05101e)

Right, I forgot about this one.

I used to have the exactly same fix in 102508 which I removed once #100488 got merged.

Cherry-picked ff2c61424598924a681e66d71e8dee to this PR.

**Comment by kolyshkin** [comment]
> is there a CI job demonstrating #102676 we can see turn green with this?

Alas I don't know how to reproduce this. It should happen when kubelet decides to create a pod cgroup twice (IOW create a pod cgroup which already exists), but I don't know kubernetes enough to pinpoint in which exact scenario it's doing this (it might be something like create two containers in a pod, remove one, then create one -- but I'm not sure). There may be no test case that does exactly that.

It was a clear regression in runc and a clear fix, so I'm 99% sure it fixes the issue #102676.

**Comment by kolyshkin** [comment]
> Keywords which can automatically close issues and at(@) or hashtag(#) mentions are not allowed in commit messages.
> 
> The list of commits with invalid commit messages:
> 
> 0dae379 Switch to non-deprecated timestamppb.Now()

Fixed.

**Comment by liggitt** [comment]
/hold

until https://github.com/kubernetes/kubernetes/pull/102508#issuecomment-873401653 https://github.com/kubernetes/kops/issues/11911 is understood so we know we're not backporting a worse regression than the one this is fixing

**Comment by odinuge** [comment]
Upstream PR was reverted in https://github.com/kubernetes/kubernetes/pull/103483. This should not be cherry-picked since it introduces an even worse regression..

/close

**Pull request closed without merging**